### PR TITLE
Add PointSourceDetections and Exposures tables

### PR DESCRIPTION
**Pull request by spenczar**
This is a sketch of an idea, more than a solid proposal. I think this would be useful in many, many places.

One question is what the fields of PointSourceDetections should be. Should it hold SphericalCoordinates, actually? Kinda weird, but formally right. It'd have nulls for so many of the values of the table, in practice (rho, vrho, vlon, vlat...). Yeah, radio exists, but it's pretty exceptional for our work.

But maybe there are big advantages to standardizing on SphericalCoordinates here, I'd totally buy that. It might make transforms of the observations much easier.

### CODE REVIEW

File: adam_core/observations/exposures.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        unique_codes = self.observatory_code.unique()
+        for code in unique_codes:
+            mask = pc.equal(self.observatory_code, code)
+            yield code, self.apply_mask(mask)
+
+    def observer_states(
+        self,
+        frame: Literal["ecliptic", "equatorial"] = "ecliptic",
+        origin: origin.OriginCodes = origin.OriginCodes.SUN,
+    ) -> cartesian.CartesianCoordinates:
```
Comment (spenczar) [inline adam_core/observations/exposures.py:40]: I was *really* feeling the power of adam_core when I wrote this method. Nice job!

File: adam_core/observations/detections.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        Returns an iterator of PointSourceDetections, each grouped by healpixel.
+        """
+        # Gather unique healpixels
+        healpixels = self.healpixels(nside, nest)
+        unique_healpixels = np.unique(healpixels)
+        for healpixel in unique_healpixels:
+            mask = pyarrow.compute.equal(healpixels, healpixel)
+            yield (healpixel, self.apply_mask(mask))
+
+    def link_to_exposures(
```
Comment (moeyensj) [inline adam_core/observations/detections.py:73]: This is clean and is something I'll likely imitate for linkage members and similar constructs.

**Review by moeyensj — APPROVED** [review]
I'm happy with this. I think we should merge this and play around with it in downstream codes and adjust as necessary.

File: adam_core/observations/detections.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    id = qv.StringColumn()
+
+    exposure_id = qv.StringColumn(nullable=True)
+
+    # Some, but not all, point source data may include times for
+    # individual observations within an exposure.
+    time = times.Times.as_column(nullable=True)
+
+    # Should this be a SphericalCoordinates instead?
```
[the remark this answers is not in the document]
Comment (spenczar) [reply adam_core/observations/detections.py:31]: I think a `to_spherical_coordinates()` method sounds clearly good. I think we'll know what we want pretty quickly once we get going on range and shift.

**Pull request merged**